samples/drivers/touch/src/main.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2024 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sample, LOG_LEVEL_INF);
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/drivers/display.h>
+
+#define CROSS_WIDTH  20
+#define CROSS_HEIGHT 20
+#define BUFFER_SIZE  ((CROSS_WIDTH) * (CROSS_HEIGHT) * 4)
+#define REFRESH_RATE 100
+
+#ifndef DT_CHOSEN_zephyr_touch
+#error "Unsupported board: zephyr,touch is not assigned"
+#endif
+
+#ifndef DT_CHOSEN_zephyr_display
+#error "Unsupported board: zephyr,display is not assigned"
+#endif
+
+static const struct device *const display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
+static const struct device *const touch_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_touch));
+static struct display_capabilities capabilities;
+static struct display_buffer_descriptor buf_desc = {.buf_size = BUFFER_SIZE,
+						    .pitch = CROSS_WIDTH,
+						    .width = CROSS_WIDTH,
+						    .height = CROSS_HEIGHT};
+
+static uint8_t buffer_cross[BUFFER_SIZE] = {0};
+static const uint8_t buffer_cross_empty[BUFFER_SIZE] = {0};
```

Comment:
drop the initializers, these are already zero initialized automatically

Reply:
Zero initializers were dropped